scripts/functions/models_transcript.py

```diff
@@ -0,0 +1,95 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import torch
+from torch.utils.data import DataLoader
+import pandas as pd
+import numpy as np
+
+
+def weights_init_normal(m):
+    classname = m.__class__.__name__
+    if classname.find("Conv") != -1:
+        torch.nn.init.normal_(m.weight.data, 0.0, 0.02)
+        if hasattr(m, "bias") and m.bias is not None:
+            torch.nn.init.constant_(m.bias.data, 0.0)
+    elif classname.find("BatchNorm2d") != -1:
+        torch.nn.init.normal_(m.weight.data, 1.0, 0.02)
+        torch.nn.init.constant_(m.bias.data, 0.0)
+
+
+##############################
+#           RESNET
+##############################
+
+
+class ResidualBlock(nn.Module):
+    def __init__(self, in_features):
+        super(ResidualBlock, self).__init__()
+
+        self.block = nn.Sequential(
+            nn.Linear(in_features, in_features),
+            nn.ReLU(),
+            nn.Linear(in_features, in_features)
+        )
+
+    def forward(self, x):
+        return x + self.block(x)
+
+
+class GeneratorResNet(nn.Module):
+    def __init__(self, input_shape, hidden_dim, output_dim, num_residual_blocks):
+        super(GeneratorResNet, self).__init__()
+
+        input_dim = input_shape[1]
+
+        # Encoder (Downsampling)
+        model = [
```

Comment:
For your model do you need to do this down sampling/resnet block stuff? Could you get away with a simple fully connected network such as:
```
nn.Linear(input_dim, encoding_dim),
nn.ReLU(),
nn.Linear(encoding_dim, hidden_dim),
nn.ReLU(),
nn.Linear(hidden_dim, input_dim)
```
or something of that variety?

Reply:
So this is misleading.  I did change the script to use a fully connected block instead of this resnet, but I kept original function names and comments.  But there is an encoder --> fully connected layer-->decoder